glusterd2/xlator/options/options.go

```diff
@@ -70,6 +82,164 @@ type Option struct {
+		err = ValidateTime(o, val)
+	case OptionTypeXlator:
+		err = ValidateXlator(o, val)
+	default:
+		err = ValidateOption(o, val)
+	}
+	return err
+}
+
+// ValidateBool validates if the option is of type boolean
+func ValidateBool(o *Option, val string) error {
+	if val == "" {
+		err := errors.New("No argument passed")
+		return err
+	}
+	switch strings.ToLower(val) {
+	case "on", "yes", "true", "enable", "0", "off", "no", "false", "disable", "1":
+		return nil
+	default:
+		return ErrInvalidArg
+	}
+}
+
+// ValidateClientAuthAddr validates mount auth address
+func ValidateClientAuthAddr(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+//ValidateDouble validates if the option is of type double
+func ValidateDouble(o *Option, val string) error {
+	if validate.IsFloat(val) != true {
+		return ErrInvalidArg
+	}
 	return nil
 }
+
+// ValidateInt validates if the option is of type Int
+func ValidateInt(o *Option, val string) error {
+	if validate.IsInt(val) != true {
+		return ErrInvalidArg
+	}
+	return nil
+}
+
+// ValidateInternetAddress validates the Internet Address
+func ValidateInternetAddress(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidateInternetAddressList validates the Internet Address List
+func ValidateInternetAddressList(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidatePath validates if the option is a valid path
+func ValidatePath(o *Option, val string) error {
+	var err error
+	t, _ := validate.IsFilePath(val)
+	if t != true {
+		err = errors.New("invalid path given")
+	} else if filepath.IsAbs(val) != true {
+		err = errors.New("option is not an absolute path name")
+	}
+	return err
+}
+
+// ValidatePercent validates if the option is in correct percent format
+func ValidatePercent(o *Option, val string) error {
+	var percent float64
+	var err error
+	l := len(val)
+	if val[0] == '%' {
+		err = ErrInvalidArg
+	} else if val[l-1] == '%' {
+		percent, err = strconv.ParseFloat(val[:(l-1)], 64)
+	} else {
+		percent, err = strconv.ParseFloat(val, 64)
+	}
+	if percent < 0.0 || percent > 100.0 {
+		err = errors.New("option is out of range [0 - 100]")
+	}
+	return err
+}
+
+// ValidatePercentOrSize validates either a correct percent format or size
+func ValidatePercentOrSize(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidatePriorityOrSize validates either priority or size
+func ValidatePriorityOrSize(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidateSizeList validates if the option is a valid size list
+func ValidateSizeList(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidateSizet validates if the option is a valid size
+func ValidateSizet(o *Option, val string) error {
+	return ErrInvalidArg
+}
+
+// ValidateStr validates if the option is of type Str
+func ValidateStr(o *Option, val string) error {
+	l := len(val)
+	if l == 0 {
+		return ErrInvalidArg
+	} else if len(strings.TrimSpace(val)) == 0 {
+		return ErrInvalidArg
+	}
```

Comment:
If `o.Value` is not empty, you need to check if the given value is among the valid values.

Reply:
@kshlm rectified, please check :)